tests/python/test_static.py

```diff
@@ -1,43 +1,44 @@
 import taichi as ti
 import numpy as np
+import pytest
 
 
-@ti.all_archs
-def test_static_if():
-    for val in [0, 1]:
-        ti.init()
-        x = ti.field(ti.i32)
+@pytest.mark.parametrize('val', [0, 1])
+@ti.test(ti.cpu)
+def test_static_if(val):
+    x = ti.field(ti.i32)
 
-        ti.root.dense(ti.i, 1).place(x)
+    ti.root.dense(ti.i, 1).place(x)
 
-        @ti.kernel
-        def static():
-            if ti.static(val > 0.5):
-                x[0] = 1
-            else:
-                x[0] = 0
+    @ti.kernel
+    def static():
+        if ti.static(val > 0.5):
+            x[0] = 1
+        else:
+            x[0] = 0
 
-        static()
-        assert x[0] == val
+    static()
+    assert x[0] == val
 
 
-@ti.must_throw(AssertionError)
+@ti.test(ti.cpu)
 def test_static_if_error():
     x = ti.field(ti.i32)
 
     ti.root.dense(ti.i, 1).place(x)
 
     @ti.kernel
-    def static():
-        if ti.static(x[0]):
+    def static(val: float):
+        if ti.static(val > 0.5):
```

Comment:
It seems that a different thing is tested here, which doesn't harm but it would be nice to have a separate test case for that. Please add another test case instead of modifying an existing one.

Reply:
The original one raises `AssertionError: __getitem__ cannot be called in taichi scope` cause x[0] is not transformed into ti.subscript(x, 0)

Reply:
Btw, I think we can actually merge `x.subscript` with `x.__getitem__` **by distinguishing `impl.inside_kernel()`**, so that such transform is no longer required?

Reply:
```suggestion
    def static():
        if ti.static(x.subscript(0)):
```
FYI you must use this to make it raise the correct error, if I don't modify to use an argument `val`.

Reply:
> FYI you must use this to make it raise the correct error, if I don't modify to use an argument `val`.

OK, sounds good!



> Btw, I think we can actually merge `x.subscript` with `x.__getitem__` **by distinguishing `impl.inside_kernel()`**, so that such transform is no longer required?

Sound like a more systematic solution is https://github.com/taichi-dev/taichi/issues/1426.